app/code/Magento/Cron/Observer/ProcessCronQueueObserver.php

```diff
@@ -56,6 +56,16 @@ class ProcessCronQueueObserver implements ObserverInterface
      */
     const SECONDS_IN_MINUTE = 60;
 
+    /**
+     * How long to wait for cron group to become unlocked
+     */
+    const LOCK_TIMEOUT = 5;
```

Comment:
I would not hard-code timeout. Some processes can take more than 5min based on the amount of the processed data. The easiest solution would be to make it configurable (either in Admin Panel or in config.php). Better solution could be to adjust automatically (e.g., make each process output something and if there is no output consider it hung and release the lock), but it might be a more complex solution.

Reply:
This is actually five seconds, after that we give up and continue to the next group.

Reasoning being: *cron:run* is triggered from system crontab every minute, so next check will happen within so small period, that it does not make sense to wait for a long time. I suppose this could be reduced to even one or zero seconds (skip group immediately if lock could not be taken).

Reply:
Makes sense. Thanks. I think it's ok to leave it as is (5s).